### PR TITLE
flow configuration examples

### PR DESCRIPTION
**Pull request by bshifter**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### What's in this PR?
logging flow configuration examples

### To Do
replace labels with the new logging-extensions image

### CODE REVIEW

File: docs/logging-extensions/_index.md [hunk 1]
```diff
@@ -61,54 +61,40 @@ EOF
...
+| controlNamespace | string | Yes | - | The resources of Eventtailer will be placed into this namespace<br> |
 | positionVolume | volume.KubernetesVolume | No | - | Volume definition for tracking fluentbit file positions (optional)<br> |
 | workloadMetaOverrides | *types.MetaBase | No | - | Override metadata of the created resources<br> |
 | workloadOverrides | *types.PodSpecBase | No | - | Override podSpec fields for the given statefulset<br> |
 | containerOverrides | *types.ContainerBase | No | - | Override container fields for the given statefulset<br> |
 
-## Kubernetes Host Tailers
-
-Tailing logs from the nodes like `kubelet`, `audit` logs or from the `systemd` journal.
+### Example: Configure logging Flow to listen Eventtailer
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
### Example: Configure logging Flow to route logs from an event tailer
```

File: docs/logging-extensions/_index.md [hunk 2]
```diff
@@ -61,54 +61,40 @@ EOF
...
-      path: /var/log/nginx/error.log
-      disabled: false
-  systemdTailers:
-    - name: my-systemd-tailer
-      disabled: false
-      maxEntries: 100
-      systemdFilter: kubelet.service
+  filters:
+  - tag_normaliser: {}
+  - stdout: {}
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 3]
```diff
@@ -61,54 +61,40 @@ EOF
...
+  match:
+  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
+  - select:
+      labels:
+        app.kubernetes.io/name: sample-event-tailer
+  outputRefs:
+    - null-output
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
    - sample-output
```

File: docs/logging-extensions/_index.md [hunk 4]
```diff
@@ -61,54 +61,40 @@ EOF
...
-      maxEntries: 100
-      systemdFilter: kubelet.service
+  filters:
+  - tag_normaliser: {}
+  - stdout: {}
+  match:
+  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
+  - select:
+      labels:
+        app.kubernetes.io/name: sample-event-tailer
```
Comment (pepov) [inline docs/logging-extensions/_index.md:87]:
```suggestion
        app.kubernetes.io/name: sample-eventtailer
```

File: docs/logging-extensions/_index.md [hunk 5]
```diff
@@ -61,54 +61,40 @@ EOF
...
 | workloadOverrides | *types.PodSpecBase | No | - | Override podSpec fields for the given statefulset<br> |
 | containerOverrides | *types.ContainerBase | No | - | Override container fields for the given statefulset<br> |
 
-## Kubernetes Host Tailers
-
-Tailing logs from the nodes like `kubelet`, `audit` logs or from the `systemd` journal.
+### Example: Configure logging Flow to listen Eventtailer
 
-### Example: configuration Kubernetes host tailer with multiple tailers
+The following example uses the flow's match term to listen the previously created "sample" Eventtailer's log.
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
The following example configures a flow to route the previously created `sample-eventtailer` EventTailer's log. 
```

File: docs/logging-extensions/_index.md [hunk 6]
```diff
@@ -178,6 +161,109 @@ EOF
 | path | string | No | - | Path to the loggable file<br> |
 | disabled | bool | No | - | Disable tailing the file<br> |
 
+### Example: Configure logging Flow to listen Hosttailer
+
+The following example uses the flow's match term to listen the previously created "sample" Hosttailer's log.
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
The following example uses the flow's match term to route the previously created "sample" HostTailer's log. 
```

File: docs/logging-extensions/_index.md [hunk 7]
```diff
@@ -178,6 +161,109 @@ EOF
...
+kubectl apply -f - <<EOF
+apiVersion: logging.banzaicloud.io/v1beta1
+kind: Flow
+metadata:
+  name: hosttailer-flow
+  namespace: default
+spec:
+  filters:
+  - tag_normaliser: {}
+  - stdout: {}
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 8]
```diff
@@ -178,6 +161,109 @@ EOF
...
+  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
+  match:
+  - select:
+      labels: 
+        app.kubernetes.io/name: sample-tailer
+      # there might be a need to match on container name too (in case of multiple containers)
+      container_names:
+        - sample-tailer-nginx-access
+  outputRefs:
+    - null-output
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
    - sample-output
```

File: docs/logging-extensions/_index.md [hunk 9]
```diff
@@ -178,6 +161,109 @@ EOF
 | path | string | No | - | Path to the loggable file<br> |
 | disabled | bool | No | - | Disable tailing the file<br> |
 
+### Example: Configure logging Flow to listen Hosttailer
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
### Example: Configure logging Flow to route host logs
```

File: docs/logging-extensions/_index.md [hunk 10]
```diff
@@ -178,6 +161,109 @@ EOF
...
+  namespace: default
+spec:
+  filters:
+  - tag_normaliser: {}
+  - stdout: {}
+  # keeps data matching to label, the rest of the data will be discarded by this flow implicitly
+  match:
+  - select:
+      labels: 
+        app.kubernetes.io/name: sample-tailer
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
        app.kubernetes.io/name: sample
```

File: docs/logging-extensions/_index.md [hunk 11]
```diff
@@ -178,6 +161,109 @@ EOF
...
+      # there might be a need to match on container name too (in case of multiple containers)
+      container_names:
+        - sample-tailer-nginx-access
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
        - nginx-access
```

File: docs/logging-extensions/_index.md [hunk 12]
```diff
@@ -178,6 +161,109 @@ EOF
...
+  outputRefs:
+    - null-output
+EOF
+```
+
+### Example: configuration Kubernetes host tailer with multiple tailers
```
Comment (pepov) [inline docs/logging-extensions/_index.md:187]:
```suggestion
### Example: configure multiple host tailers
```

File: docs/logging-extensions/_index.md [hunk 13]
```diff
@@ -178,6 +161,109 @@ EOF
...
+kubectl apply -f - <<EOF
+apiVersion: logging-extensions.banzaicloud.io/v1alpha1
+kind: HostTailer
+metadata:
+  name: multi-sample
+spec:
+  # resources will be placed in default namespace
+  controlNamespace: default
+  # recreate resources when it's unable to change a field due to immutable status
+  enableRecreateWorkloadOnImmutableFieldChange: true
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 14]
```diff
@@ -178,6 +161,109 @@ EOF
...
+  # list of File tailers
+  fileTailers:
+    - name: nginx-access
+      path: /var/log/nginx/access.log
+      disabled: false
+    - name: nginx-error
+      path: /var/log/nginx/error.log
+      # currently unwanted
+      disabled: true
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 15]
```diff
@@ -178,6 +161,109 @@ EOF
...
+spec:
+  # resources will be placed in default namespace
+  controlNamespace: default
+  # recreate resources when it's unable to change a field due to immutable status
+  enableRecreateWorkloadOnImmutableFieldChange: true
+  # list of File tailers
+  fileTailers:
+    - name: nginx-access
+      path: /var/log/nginx/access.log
+      disabled: false
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 16]
```diff
@@ -178,6 +161,109 @@ EOF
...
+    - name: nginx-error
+      path: /var/log/nginx/error.log
+      # currently unwanted
+      disabled: true
+  # list of Systemd tailers
+  systemdTailers:
+    - name: my-systemd-tailer
+      disabled: false
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 17]
```diff
@@ -178,6 +161,109 @@ EOF
...
+
+| Variable Name | Type | Required | Default | Description |
+|---|---|---|---|---|
+| fileTailers | []FileTailer | No | - | List of file tailers<br> |
+| systemdTailers | []SystemdTailer | No | - | List of systemd tailers<br> |
+| enableRecreateWorkloadOnImmutableFieldChange | bool | No | - | EnableRecreateWorkloadOnImmutableFieldChange enables the operator to recreate the<br>fluentbit daemonset and the fluentd statefulset (and possibly other resource in the future)<br>in case there is a change in an immutable field<br>that otherwise couldn't be managed with a simple update.<br> |
+| controlNamespace | string | Yes | - | The resources of HostTailer will be placed into this namespace<br> |
+| workloadMetaOverrides | *types.MetaBase | No | - | Override metadata of the created resources<br> |
+| workloadOverrides | *types.PodSpecBase | No | - | Override podSpec fields for the given daemonset<br> |
+| containerOverrides | *types.ContainerBase | No | - | Override container fields for the given daemonset<br> |
```
Comment (pepov) [inline docs/logging-extensions/_index.md]:
```suggestion
```

File: docs/logging-extensions/_index.md [hunk 18]
```diff
@@ -178,6 +161,109 @@ EOF
...
+
+### Example: Configure logging Flow with explicit exclusion
```
Comment (pepov) [inline docs/logging-extensions/_index.md]: we don't need this example, log routing is discussed in detail in the logging-operator docs

**Review by pepov — CHANGES_REQUESTED** [review]

File: docs/logging-extensions/_index.md (hunk 4 above)
Comment (bshifter) [reply docs/logging-extensions/_index.md:87]: // EventTailer configuration
var EventTailer = EventTailerConfig{
	TailerAffix:           "event-tailer",
	ConfigurationFileName: "config.json",
	PositionVolumeName:    "event-tailer-position",
}

Comment (bshifter) [reply docs/logging-extensions/_index.md:87]: the label name will be "sample-event-tailer" with the current configuration

**Review by pepov — APPROVED** [review]

**Pull request merged**